src/main/java/net/mcreator/ui/help/HelpLoader.java

```diff
@@ -84,42 +84,44 @@ public static int getCoverageForLocale(Locale locale) {
 	}
 
 	public static boolean hasFullHelp(IHelpContext helpContext) {
-		return helpContext != null && helpContext.getEntry() != null && getFromCache(helpContext.getEntry()) != null;
+		return helpContext != null && helpContext.entry() != null && getFromCache(helpContext.entry()) != null;
 	}
 
 	public static String loadHelpFor(IHelpContext helpContext) {
 		if (helpContext != null) {
 			URI uri = null;
 			try {
-				uri = helpContext.getContextURL();
+				uri = helpContext.contextURL();
 			} catch (URISyntaxException ignored) {
 			}
 
 			StringBuilder helpString = new StringBuilder(
 					"<html><head><style>table{ border-collapse: collapse; border-spacing: 0; } "
 							+ "th { border: 1px solid #a0a0a0; } td { border: 1px solid #a0a0a0; } </style></head><body>");
 
-			if (helpContext.getEntry() != null) {
-				String helpText = getFromCache(helpContext.getEntry());
+			if (helpContext.entry() != null) {
+				String helpText = getFromCache(helpContext.entry());
 				if (helpText != null) {
-					if ((helpText.contains("${") || helpText.contains("<#"))
-							&& helpContext instanceof ModElementHelpContext meHelpContext) {
+					if (helpText.contains("${") || helpText.contains("<#")) {
 						try {
 							Map<String, Object> dataModel = new HashMap<>();
-							dataModel.put("data", meHelpContext.getModElementFromGUI());
-							dataModel.put("registryname",
-									meHelpContext.getModElementFromGUI().getModElement().getRegistryName());
-							dataModel.put("name", meHelpContext.getModElementFromGUI().getModElement().getName());
-							dataModel.put("elementtype",
-									meHelpContext.getModElementFromGUI().getModElement().getType().getReadableName());
 							dataModel.put("l10n", new L10N());
 
-							if (meHelpContext.getModElementFromGUI().getModElement().getGenerator() != null)
-								dataModel.putAll(meHelpContext.getModElementFromGUI().getModElement().getGenerator()
-										.getBaseDataModelProvider().provide());
-
-							Template freemarkerTemplate = new Template(helpContext.getEntry(),
-									new StringReader(helpText), configuration);
+							if (helpContext instanceof ModElementHelpContext meHelpContext) {

```

Comment:
Why did you move check down here and remove generator null check?

Reply:
1. This way tips created from `IHelpContext.NONE` are able to use L10N for translations;
2. I didn't remove any code, WDYM certainly?

Reply:
Ok saw it few lines lower, my bad :D